src/mono/mono/metadata/image.c

```diff
@@ -1846,34 +1846,70 @@ mono_image_loaded_by_guid (const char *guid)
 	return mono_image_loaded_by_guid_internal (guid, FALSE);
 }
 
+#ifdef ENABLE_NETCORE
+static const char *
+mono_get_image_culture (MonoImage *image)
+{
+	MonoTableInfo *t = &image->tables [MONO_TABLE_ASSEMBLY];
+	if (!t->rows)
+		return NULL;
+
+	guint32 cols [MONO_ASSEMBLY_SIZE];
+	mono_metadata_decode_row (t, 0, cols, MONO_ASSEMBLY_SIZE);
+	return mono_metadata_string_heap (image, cols [MONO_ASSEMBLY_CULTURE]);
+}
+#endif
+
+char *
+mono_image_get_name_with_culture_if_needed (MonoImage *image)
+{
+#ifndef ENABLE_NETCORE
+	return NULL;
+#else
+	if (!g_str_has_prefix (image->name, "data-") &&
+		!g_path_is_absolute (image->name))
+	{
+		const char *culture = mono_get_image_culture (image);
+
+		if (culture && culture[0] != 0)
+			return g_strdup_printf ("(%s)%s", culture, image->name);
```

Comment:
Shouldn't we just do `%s/%s`? I did confirm that `/` is not allowed in the simple name otherwise.

Reply:
Sounds good. I wasn't sure if we wanted to use `/` at the beginning. Thanks for checking.